### PR TITLE
use PasswordString object

### PR DESCRIPTION
**Pull request by colombod**
using PasswordString object in GetPAsswordGesture to prevent printing password values in clear

image

### CODE REVIEW

File: src/Microsoft.DotNet.Interactive/PasswordString.cs [hunk 1]
```diff
@@ -1,8 +1,13 @@
 ﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using Microsoft.DotNet.Interactive.Formatting;
```
Comment (colombod) [inline src/Microsoft.DotNet.Interactive/PasswordString.cs:4]: This is not needed

File: src/Microsoft.DotNet.Interactive/PasswordStringFormatterSource.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Text.Json;
+using Microsoft.DotNet.Interactive.Formatting;
+
+namespace Microsoft.DotNet.Interactive;
+
+internal class PasswordStringFormatterSource : ITypeFormatterSource
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive/PasswordStringFormatterSource.cs:10]: This seems more complex than needed. A custom `ToString` should work fine.

Comment (colombod) [inline src/Microsoft.DotNet.Interactive/PasswordStringFormatterSource.cs:10]: not needed

File: src/Microsoft.DotNet.Interactive/PasswordStringJsonConverter.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Text.Json;
+
+namespace Microsoft.DotNet.Interactive;
+
+internal class PasswordStringJsonConverter : JsonConverter<PasswordString>
+{
+    public override PasswordString Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
```
Comment (colombod) [inline src/Microsoft.DotNet.Interactive/PasswordStringJsonConverter.cs:11]: remove this

File: src/Microsoft.DotNet.Interactive/PasswordStringJsonConverter.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            }
+        }
+
+        throw new JsonException($"Cannot deserialize {typeToConvert.Name}");
+    }
+
+    public override void Write(Utf8JsonWriter writer, PasswordString value, JsonSerializerOptions options)
+    {
+        writer.WriteStartObject();
+        writer.WriteString("password", value.GetClearTextPassword());
```
Comment (shyamnamboodiripad) [inline src/Microsoft.DotNet.Interactive/PasswordStringJsonConverter.cs:49]: Is it by design that passwords flow through the system as clear text (across process boundary)? Or is this a known point in time issue that we plan to address later.

**Review by shyamnamboodiripad — APPROVED** [review]

File: src/Microsoft.DotNet.Interactive/PasswordStringJsonConverter.cs (hunk 4 above)
Comment (jonsequitur) [reply src/Microsoft.DotNet.Interactive/PasswordStringJsonConverter.cs:49]: It's by design. Encrypting data in transit is the responsibility of transport-level security as passwords aren't the only sensitive information we can assume people are sending and receiving.

With than in mind, this PR is intended only to obfuscate certain pieces of information from display in the UI.

**Pull request merged**